packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

```diff
@@ -284,12 +284,16 @@ contract Deploy is Deployer {
         dii.set(dii.mipsVersion.selector, Config.useMultithreadedCannon() ? 2 : 1);
         string memory release = "dev";
         dii.set(dii.l1ContractsRelease.selector, release);
-        dii.set(dii.superchainConfigProxy.selector, artifacts.mustGetAddress("SuperchainConfigProxy"));
         dii.set(dii.protocolVersionsProxy.selector, artifacts.mustGetAddress("ProtocolVersionsProxy"));
-        dii.set(
-            dii.upgradeController.selector,
-            IProxyAdmin(EIP1967Helper.getAdmin(artifacts.mustGetAddress("SuperchainConfigProxy"))).owner()
-        );
+
+        ISuperchainConfig superchainConfig = ISuperchainConfig(artifacts.mustGetAddress("SuperchainConfigProxy"));
+        dii.set(dii.superchainConfigProxy.selector, artifacts.mustGetAddress("SuperchainConfigProxy"));
+
+        address superchainProxyAdmin = EIP1967Helper.getAdmin(address(superchainConfig));
+        dii.set(dii.superchainProxyAdmin.selector, superchainProxyAdmin);
+
+        // I think this was a bug
+        dii.set(dii.upgradeController.selector, IProxyAdmin(superchainProxyAdmin).owner());
```

Comment:
we should make sure to add a test that would have caught this bug

Reply:
Yes, that's the regression test I mentioned planning to do

Reply:
ahhh got it, that did not click lol

Reply:
I actually don't think this was a bug anymore. I though that the upgrade controller was being set to the ProxyAdmin, rather than the ProxyAdminOwner.

ie.  my brain made the following substitution error:
```diff
dii.set(
  dii.upgradeController.selector,
-  IProxyAdmin(EIP1967Helper.getAdmin(artifacts.mustGetAddress("SuperchainConfigProxy"))).owner()
+  IProxyAdmin(EIP1967Helper.getAdmin(artifacts.mustGetAddress("SuperchainConfigProxy")))
);
```

But I'm removing doubt by beefing up `ChainAssertions.checkOpContractsManager()`: a2b8101c727d